### PR TITLE
Remove empty dirs when cleaning with Dir opt.

### PR DESCRIPTION
**Pull request by kuba--**
Signed-off-by: kuba-- <kuba@sourced.tech>
Fixes https://github.com/src-d/go-git/issues/895

When `CleanOptions.Dir` is set on `true` then an empty dir (with subdirs) could be removed (this is what `git clean -f -d .` does).

### CODE REVIEW

**Comment by kuba--** [comment]
@onodera-punpun - Could you check if this fix works for you?

**Comment by ajnavarro** [comment]
@kuba-- could you add a test reproducing the error on #895, please?

**Comment by kuba--** [comment]
@ajnavarro - done. I've added a test to check if empty dirs are cleaned, as well.

File: worktree.go [hunk 1]
```diff
@@ -713,30 +713,65 @@ func (w *Worktree) readGitmodulesFile() (*config.Modules, error) {
...
 		return err
 	}
 
-	// Check Worktree status to be Untracked, obtain absolute path and delete.
-	for relativePath, status := range s {
-		// Check if the path contains a directory and if Dir options is false,
-		// skip the path.
-		if relativePath != filepath.Base(relativePath) && !opts.Dir {
-			continue
+	var clean func(string, []os.FileInfo) error
```
Comment (mcuadros) [inline worktree.go]: can you move this function to a member function called `doClean`

File: worktree.go [hunk 2]
```diff
@@ -713,30 +713,65 @@ func (w *Worktree) readGitmodulesFile() (*config.Modules, error) {
...
+					}
+				}
+			}
 		}
 
-		// Remove the file only if it's an untracked file.
-		if status.Worktree == Untracked {
-			absPath := filepath.Join(w.Filesystem.Root(), relativePath)
-			if err := os.Remove(absPath); err != nil {
+		if opts.Dir {
```
Comment (mcuadros) [inline worktree.go]: also will be nice move this code to a function something like `doCleanDirectories`

**Comment by kuba--** [comment]
Why randomly this test on windows fails?
```
----------------------------------------------------------------------
FAIL: <autogenerated>:1: UploadPackSuite.TearDownSuite

c:/gopath/src/gopkg.in/src-d/go-git-fixtures.v3/fixtures.go:368:
    c.Assert(Clean(), check.IsNil)
... value *os.PathError = &os.PathError{Op:"remove", Path:"C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\tmp-tgz-799335987\\objects\\pack\\pack-a3fed42da1e8189a077c0e6846c040dcf73fc9dd.idx", Err:0x5} ("remove C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\tmp-tgz-799335987\\objects\\pack\\pack-a3fed42da1e8189a077c0e6846c040dcf73fc9dd.idx: Access is denied.")

OOPS: 31 passed, 7 skipped, 1 FAILED
--- FAIL: Test (8.31s)
```

**Comment by kuba--** [comment]
@mcuadros - PR refined.

**Pull request merged**